pkg/verifier/codeowner.go

```diff
@@ -0,0 +1,213 @@
+			if err != nil || !exists {
+				exists, err = p.GroupExists(owner)
+				if err != nil || !exists {
+					valid = false
+					log.Errorf("Error parsing line %d: user/group %s is invalid", c.Line, element)
+				}
+			}
+		}
+	}
+	return valid, nil
+}
+
+// getPatternFromLine converts a line to a CODEOWNERS entry
+// This is roughly addapted from https://github.com/sabhiram/go-gitignore
+func getPatternFromLine(lineNumber string) (*regexp.Regexp, bool) {
+	// Trim OS-specific carriage returns.
+	lineNumber = strings.TrimRight(lineNumber, "\r")
+
+	// TODO: Handle [Rule 4] which negates the match for patterns leading with "!"
+	negatePattern := false
+	if lineNumber[0] == '!' {
+		negatePattern = true
+		lineNumber = lineNumber[1:]
+	}
+
+	// If we encounter a foo/*.blah in a folder, prepend the / char
+	if regexp.MustCompile(`([^\/+])/.*\*\.`).MatchString(lineNumber) && lineNumber[0] != '/' {
+		lineNumber = "/" + lineNumber
+	}
+
+	// Handle escaping the "." char
+	lineNumber = regexp.MustCompile(`\.`).ReplaceAllString(lineNumber, `\.`)
+
+	magicStar := "#$~"
+
+	// Handle "/**/" usage
+	if strings.HasPrefix(lineNumber, "/**/") {
+		lineNumber = lineNumber[1:]
+	}
+	lineNumber = regexp.MustCompile(`/\*\*/`).ReplaceAllString(lineNumber, `(/|/.+/)`)
+	lineNumber = regexp.MustCompile(`\*\*/`).ReplaceAllString(lineNumber, `(|.`+magicStar+`/)`)
+	lineNumber = regexp.MustCompile(`/\*\*`).ReplaceAllString(lineNumber, `(|/.`+magicStar+`)`)
+
+	// Handle escaping the "*" char
+	lineNumber = regexp.MustCompile(`\\\*`).ReplaceAllString(lineNumber, `\`+magicStar)
+	lineNumber = regexp.MustCompile(`\*`).ReplaceAllString(lineNumber, `([^/]*)`)
+
+	// Handle escaping the "?" char
+	lineNumber = strings.Replace(lineNumber, "?", `\?`, -1)
+
+	lineNumber = strings.Replace(lineNumber, magicStar, "*", -1)
+
+	// Temporary regex
+	var expr = ""
+	if strings.HasSuffix(lineNumber, "/") {
+		expr = lineNumber + "(|.*)$"
+	} else {
+		expr = lineNumber + "(|/.*)$"
+	}
+	if strings.HasPrefix(expr, "/") {
+		expr = "^(|/)" + expr[1:]
+	} else {
+		expr = "^(|.*/)" + expr
+	}
+	pattern, _ := regexp.Compile(expr)
+
+	return pattern, negatePattern
+}
+
+// MatchesPath returns true if the given GitIgnore structure would target
+// a given path string `f`.
+func (co *CodeOwner) MatchesPath(f string) bool {
+	// Replace OS-specific path separator.
+	f = strings.Replace(f, string(os.PathSeparator), "/", -1)
+
+	matchesPath := false
+	if co.Regex.MatchString(f) {
+		// If this is a regular target (not negated with a gitignore exclude "!" etc)
+		if !co.Negate {
+			matchesPath = true
+		}
+	}
+	return matchesPath
+}
+
+// VerifyCodeowner check if a line matches any entry on the reversed list of CodeOwners
+func VerifyCodeowner(codeowners []*CodeOwner, filename string, ignore []string) (*CodeOwner, bool) {
+	for _, c := range reverseCodeOwners(codeowners) {
+		match := c.MatchesPath(filename)
+		if match {
+			owners := difference(c.Owners, ignore)
```

Comment:
Since we don't use the output of this function, it could return a boolean and make the code more simple

```go
if match {
  return c, hasDifference(c.Owners, ignore)
  // or
  return c, !isIngnoredCodeowners(c.Owners, ignoredCodeowners)
}
```